crates/runtime/src/task_history.rs

```diff
@@ -15,124 +15,72 @@ limitations under the License.
 */
 
 use crate::accelerated_table::refresh::Refresh;
+use crate::datafusion::DataFusion;
+use crate::dataupdate::DataUpdate;
 use crate::internal_table::create_internal_accelerated_table;
 use crate::{component::dataset::acceleration::Acceleration, datafusion::SPICE_RUNTIME_SCHEMA};
 use crate::{component::dataset::TimeFormat, secrets::Secrets};
+use arrow::array::{ArrayBuilder, MapBuilder, RecordBatch, StringBuilder};
 use arrow::datatypes::{DataType, Field, Schema, TimeUnit};
+use data_components::arrow::struct_builder::StructBuilder;
 use datafusion::sql::TableReference;
+use snafu::prelude::*;
 use snafu::{ResultExt, Snafu};
-use std::fmt::Display;
+use std::collections::HashMap;
 use std::sync::Arc;
-use std::time::Duration;
+use std::time::{Duration, SystemTime};
 use tokio::sync::RwLock;
 
 use crate::accelerated_table::{AcceleratedTable, Retention};
 
-pub const DEFAULT_TASK_HISTORY_TABLE: &str = "task_history";
+pub mod otel_exporter;
 
-pub enum TaskType {
-    SqlQuery,
-    NsqlQuery,
-    AiCompletion,
-    TextEmbed,
-    VectorSearch,
-}
+pub const DEFAULT_TASK_HISTORY_TABLE: &str = "task_history";
 
-impl Display for TaskType {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        match self {
-            TaskType::SqlQuery => write!(f, "sql_query"),
-            TaskType::NsqlQuery => write!(f, "nsql_query"),
-            TaskType::AiCompletion => write!(f, "ai_completion"),
-            TaskType::TextEmbed => write!(f, "text_embed"),
-            TaskType::VectorSearch => write!(f, "vector_search"),
-        }
-    }
-}
+// pub enum TaskType {
```

Comment:
to remove?

Reply:
Yeah, it's not being used but I didn't want to remove it yet.